Source/Core/Core/IOS/USB/Bluetooth/BTReal.cpp

```diff
@@ -196,68 +196,70 @@ IPCCommandResult BluetoothReal::IOCtlV(const IOCtlVRequest& request)
+    libusb_fill_control_setup(buffer.get(), cmd->request_type, cmd->request, cmd->value, cmd->index,
+                              cmd->length);
+    Memory::CopyFromEmu(buffer.get() + LIBUSB_CONTROL_SETUP_SIZE, cmd->data_address, cmd->length);
     libusb_transfer* transfer = libusb_alloc_transfer(0);
-    transfer->flags |= LIBUSB_TRANSFER_FREE_TRANSFER;
-    libusb_fill_control_transfer(transfer, m_handle, buffer.data(), CommandCallback, cmd.release(),
-                                 0);
+    transfer->flags |= LIBUSB_TRANSFER_FREE_TRANSFER | LIBUSB_TRANSFER_FREE_BUFFER;
+    libusb_fill_control_transfer(transfer, m_handle, buffer.release(), CommandCallback,
+                                 cmd.release(), 0);
     libusb_submit_transfer(transfer);
     break;
   }
   // ACL data (incoming or outgoing) and incoming HCI events (respectively)
-  case USBV0_IOCTL_BLKMSG:
-  case USBV0_IOCTL_INTRMSG:
+  case USB::IOCTLV_USBV0_BLKMSG:
+  case USB::IOCTLV_USBV0_INTRMSG:
   {
-    auto buffer = std::make_unique<CtrlBuffer>(request);
-    if (request.request == USBV0_IOCTL_INTRMSG && m_fake_read_buffer_size_reply.TestAndClear())
+    auto buffer = std::make_unique<USB::V0IntrMessage>(request);
+    if (request.request == USB::IOCTLV_USBV0_INTRMSG)
     {
-      FakeReadBufferSizeReply(*buffer);
-      return GetNoReply();
-    }
-    if (request.request == USBV0_IOCTL_INTRMSG && m_fake_vendor_command_reply.TestAndClear())
-    {
-      FakeVendorCommandReply(*buffer);
-      return GetNoReply();
-    }
-    if (request.request == USBV0_IOCTL_INTRMSG && m_sync_button_state == SyncButtonState::Pressed)
-    {
-      Core::DisplayMessage("Scanning for Wii Remotes", 2000);
-      FakeSyncButtonPressedEvent(*buffer);
-      return GetNoReply();
-    }
-    if (request.request == USBV0_IOCTL_INTRMSG &&
-        m_sync_button_state == SyncButtonState::LongPressed)
-    {
-      Core::DisplayMessage("Reset saved Wii Remote pairings", 2000);
-      FakeSyncButtonHeldEvent(*buffer);
-      return GetNoReply();
+      if (m_sync_button_state == SyncButtonState::Pressed)
+      {
+        Core::DisplayMessage("Scanning for Wii Remotes", 2000);
+        FakeSyncButtonPressedEvent(*buffer);
+        return GetNoReply();
+      }
+      if (m_sync_button_state == SyncButtonState::LongPressed)
+      {
+        Core::DisplayMessage("Reset saved Wii Remote pairings", 2000);
+        FakeSyncButtonHeldEvent(*buffer);
+        return GetNoReply();
+      }
+      if (m_fake_read_buffer_size_reply.TestAndClear())
+      {
+        FakeReadBufferSizeReply(*buffer);
+        return GetNoReply();
+      }
+      if (m_fake_vendor_command_reply.TestAndClear())
+      {
+        FakeVendorCommandReply(*buffer);
+        return GetNoReply();
+      }
     }
     libusb_transfer* transfer = libusb_alloc_transfer(0);
-    transfer->buffer = Memory::GetPointer(buffer->m_payload_addr);
+    transfer->buffer = buffer->MakeBuffer(buffer->length).release();
```

Comment:
Oh wow, I didn't know that! Guess I'll revert this change temporarily, since #4690 will completely sidestep this issue since we will keep track of the whole transfer command.